docs/source/csdl-spec.md

```diff
@@ -0,0 +1,630 @@
+
+CSDL removes certain SDL features, as described in the [divergence](#divergence-from-graphql-sdl) section. Nevertheless, CSDL consumers **should** be prepared to parse any valid GraphQL SDL. A CSDL document which contains disallowed SDL constructs **should** trigger validation errors after parsing.
+
+CSDL introduces a suite of types and directives. All CSDL-specific types and directives are prefixed with `csdl_` as recommended in [the GraphQL specification](https://spec.graphql.org/draft/#note-ca863).
+
+### Divergence from GraphQL SDL
+Formally, CSDL is a subset of the [GraphQL schema definition language](https://spec.graphql.org/draft/#sec-Type-System).
+
+In particular, CSDL differs from GraphQL SDL in the following ways:
+- **Extensions are forbidden.** You cannot `extend type` or `extend interface` within a CSDL document.
+- **`schema @csdl(version:)` is mandatory.** Conforming CSDL documents must have a `schema` declaration which must be annotated with a `@csdl` directive specifying, at a minimum, the semver for the CSDL spec in use.
+- the CSDL must define certain types as described in the [mandatory types](#mandatory-types) section.
+
+### Mandatory Types
+
+A CSDL document **must** define a `csdl_Graph` enum. Each enum value defines a subgraph. Each value **must** be annotated with a `@csdl_endpoint` directive specifying the endpoint URL for the subgraph.
+
+```graphql
+enum csdl_Graph {
+  ROCKETS @csdl_endpoint(url: "https://rockets.api.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.api.com"),
+}
+```
+
+The `csdl_Graph` enum is used as input to csdl directives which link fields and types to subgraphs.
+
+## Scalars
+
+### `scalar csdl_SelectionSet`
+
+A GraphQL selection set represented with the same syntax as defined in [the GraphQL SDL](https://spec.graphql.org/draft/#sec-Selection-Sets).
+
+Note: Unlike the federation `_FieldSet` scalar, a `csdl_SelectionSet` must parse as a GraphQL selection set, so the selections must be surrounded by braces.
+
+### `scalar csdl_Url`
+
+A [URL](https://www.w3.org/Addressing/URL/url-spec.html).
+
+### `scalar csdl_Version`
+
+A [Semantic versioning 2.0.0](https://semver.org/spec/v2.0.0.html) verison or range.
+
+## Data Model
+TK, draws heavily from federation, a little more general.
+
+### Realized types
+
+A type is *realized* wherever a resolver from a particular *subgraph* returns that type. We use the notation `subgraph::Type` to represent `Type` realized by `subgraph` and `subgraph::Type.field` to represent the resolver for `Type.field` within `subgraph`.
+
+<a class="listing">Query with resolvers and realized types</a>
+```graphql
+query {          # resolver                    | realizes
+                 # ----------------------------|--------------
+  rockets {      # rockets::Query.rockets     -> rockets::Rocket
+    captain {    # rockets::Rocket.astronaut  -> rockets::Astronaut
```

Comment:
I wonder if it's worthwhile to distinguish the `rockets::Rocket.astronaut` resolver from the `rockets::_Entity.Rocket.astronaut`, since in practice they may have different implementations. I suspect that in theory, we would prefer that everyone have the exact same resolver for both, but I can tell you with certainty that many users don't have any authz on their `_entities` resolvers (which is a, uh, scary security concern)

Reply:
They are different. `rockets::Query._entities` isn't explicitly mentioned here, it'll be implicitly used to port `rockets::Astronaut` to `astronauts::Astronaut`.

that *should* be mentioned here, but i'm really trying to not rewrite the federation spec. :/